### PR TITLE
Update `Invoke-PolicyToBicep.ps1` Scripts to Handle `-`'s,` `'s (spaces) & `'` apostrophes + Fix #254

### LINKED ISSUE #254: 🪲 Bug Report - loadTextContent called multiple times for the same file causes bloat
## Describe the bug

custom-policy-definitions.bicep calls loadTextContent multiple times for the same file. This causes the content of the file to be put into the output file multiple times which bloats the file significantly

## To Reproduce

Steps to reproduce the behaviour:

1. Build a bicep file that uses custom-policy-definitions.bicep and take not of the size of the output file
2. Manually replace the redundant calls to loadTextContent in custom-policy-definitions.bicep by using a var instead
3. Build the new bicep file and take note of the size of the output file

## Expected behaviour

custom-policy-definitions.bicep should only load each file it needs one time and store the content in a var to prevent file bloat

### Good example

> Note the `definitionParameters` in each of the examples

```
var publicPaasEndpointsParams = json(loadTextContent('lib/policy_set_definitions/policy_set_definition_es_deny_publicpaasendpoints.parameters.json'))

... other stuff removed for clarity

libSetChildDefinitions: [
      {
        definitionReferenceID: 'ACRDenyPaasPublicIP'
        definitionID: '/providers/Microsoft.Authorization/policyDefinitions/0fdf0491-d080-4575-b627-ad0e843cba0f'
        definitionParameters: publicPaasEndpointsParams.ACRDenyPaasPublicIP.parameters
      }
      {
        definitionReferenceID: 'AFSDenyPaasPublicIP'
        definitionID: '/providers/Microsoft.Authorization/policyDefinitions/21a8cd35-125e-4d13-b82d-2e19b7208bb7'
        definitionParameters: publicPaasEndpointsParams.AFSDenyPaasPublicIP.parameters
      }
```

### Bloated Example (code today)

```
libSetChildDefinitions: [
      {
        definitionReferenceID: 'ACRDenyPaasPublicIP'
        definitionID: '/providers/Microsoft.Authorization/policyDefinitions/0fdf0491-d080-4575-b627-ad0e843cba0f'
        definitionParameters: json(loadTextContent('lib/policy_set_definitions/policy_set_definition_es_deny_publicpaasendpoints.parameters.json')).ACRDenyPaasPublicIP.parameters
      }
      {
        definitionReferenceID: 'AFSDenyPaasPublicIP'
        definitionID: '/providers/Microsoft.Authorization/policyDefinitions/21a8cd35-125e-4d13-b82d-2e19b7208bb7'
        definitionParameters: json(loadTextContent('lib/policy_set_definitions/policy_set_definition_es_deny_publicpaasendpoints.parameters.json')).AFSDenyPaasPublicIP.parameters
      }
```
## Correlation ID

A correlation ID really helps us investigate your issue further. Please provide one if possible. Details on how to find a correlation ID can be found here: [Correlation ID and support](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/deployment-history?tabs=azure-portal#correlation-id-and-support)

## Additional context

Anything else we should know to help us troubleshoot this bug? I have created a separate but highly related issue on the bicep side here https://github.com/Azure/bicep/issues/6970

### PR DESCRIPTION
**Pull request by jtracey93**
# Overview/Summary

Update `Invoke-PolicyToBicep.ps1` Scripts to Handle `-`'s,` `'s (spaces) & `'` apostrophes + Fix #254 

## This PR fixes/adds/changes/removes

1. Update `Invoke-PolicyToBicep.ps1` Scripts to Handle `-`'s,` `'s (spaces) & `'` apostrophes
2. Fix #254 

### Breaking Changes

None.

## Testing Evidence

Local testing with ISD colleague has proved successful

image

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [x] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [x] Performed testing and provided evidence.
- [ ] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [x] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

**Comment by SeSeicht** [comment]
additional Task woulb be: escape `'` in names

**Comment by jtracey93** [comment]
> additional Task woulb be: escape `'` in names

Hey @SeSeicht could you provide some more explanation or screenshots of what you mean here as I can't work it out 👍

**Comment by SeSeicht** [comment]
> > additional Task woulb be: escape `'` in names
> 
> Hey @SeSeicht could you provide some more explanation or screenshots of what you mean here as I can't work it out 👍

This for example
image

File: .github/scripts/Invoke-PolicyToBicep-China.ps1 [hunk 1]
```diff
@@ -1,136 +1,227 @@
 <#
 SUMMARY: This PowerShell script helps with the authoring of the policy definiton module for Azure China by outputting information required for the variables within the module.
-DESCRIPTION: This PowerShell script outputs the Name & Path to a Bicep strucutred .txt file named '_mc_policyDefinitionsBicepInput.txt' and '_mc_policySetDefinitionsBicepInput.txt' respectively. It also creates a parameters file for each of the policy set definitions. It also outputs the number of policies definition and set definition files to the console for easier reviewing as part of the PR process.
-AUTHOR/S: faister, jtracey93
-VERSION: 1.1
+DESCRIPTION: This PowerShell script outputs the Name & Path to a Bicep strucutred .txt file named '_mc_policyDefinitionsBicepInput.txt' (defintionsTxtFileName) and '_mc_policySetDefinitionsBicepInput.txt' (defintionsSetTxtFileName) respectively. It also creates a parameters file for each of the policy set definitions. It also outputs the number of policies definition and set definition files to the console for easier reviewing as part of the PR process.
```
Comment (jfaurskov) [inline .github/scripts/Invoke-PolicyToBicep-China.ps1:3]:
```suggestion
DESCRIPTION: This PowerShell script outputs the Name & Path to a Bicep structured .txt file named '_mc_policyDefinitionsBicepInput.txt' (defintionsTxtFileName) and '_mc_policySetDefinitionsBicepInput.txt' (defintionsSetTxtFileName) respectively. It also creates a parameters file for each of the policy set definitions. It also outputs the number of policy and policy set definition files to the console for easier reviewing as part of the PR process.
```

**Review by jfaurskov — CHANGES_REQUESTED** [review]
Other than the initial lines at the top of the scripts I have not comments

**Pull request merged**